### ISSUE TITLE
New-SafeguardAsset does not accept Appliance parameter in v2.10.399

### ISSUE THREAD

**Issue by gilesmarshall** [OneIdentity/safeguard-ps#215]
After updating to `safeguard-ps` v2.10.399, the New-SafeguardAsset cmdlet does not accept the data passed to the `-Appliance` parameter and prompts for the appliance name.  The same command and data works with v2.9.379.

`New-SafeguardAsset @NewAssetData`
`Searching for platforms with 'SUSE Linux Enterprise Server (SLES) Other'`
`Appliance: `

Where 

`$NewAssetData = @{`
   `Appliance = $ApplianceFQDN`
   `AccessToken = $AccessToken`
   `DisplayName = $Asset_Name`
   `etc...`

[edit]
Further testing shows that even when the required parameters are provided at the prompts, the asset creation still fails with this error:
`Invoke-RestMethod : Cannot bind parameter 'Uri'. Cannot convert value "https://$SGAPPLIANCE/RSTS/oauth2/token" to type "System.Uri". Error: "Invalid URI: The hostname could not be parsed."`
`At ..Safeguard-ps\2.10.399\safeguard-ps.psm1:729 char:28`
This means that I am stuck with working with the earlier versions.

**Comment by Kevin-Andrew** [comment]
@gilesmarshall 
I am a little confused by the error message string you are showing.  In particular when it says, "Cannot convert value..." and then has the following URL:
```
https://$SGAPPLIANCE/RSTS/oauth2/token
```

Note the `$SGAPPLIANCE`.  Also note that it is saying the error comes from `Safeguard-ps\2.10.399\safeguard-ps.psm1:729`.

If we look at the source code here, [line 729 of safeguard-ps.psm1](https://github.com/OneIdentity/safeguard-ps/blob/master/src/safeguard-ps.psm1#L729), we see:
```
} -Uri "https://$Appliance/RSTS/oauth2/token" -Body @"
```

Note, in this line we have `$Appliance`.

Do you know why those might be different?  I see your path as `Safeguard-ps\2.10.399`, so I also checked the 2.10 branch, but it also has just `$Appliance` as the variable in the URL string.

So I'm not sure where that might be coming from or even if it is the problem.  We will try to take a look at what got published versus what we are seeing in the source code.

**Comment by Kevin-Andrew** [comment]
@gilesmarshall 
I have downloaded/installed the official release of the Safeguard-ps module to my machine, and then manually inspecting the source code:
![image](https://user-images.githubusercontent.com/22459191/70263528-d3b80b80-1753-11ea-9113-a332be22b171.png)

As you can see, 2.10.399 is the same version that is listed in your error message.  But when looking at line 729 of the safeguard-ps.psm1 file, I have `$Appliance` as the variable name in the URL string.

Do you think you can manually open the file on your machine and do a similar check?

**Comment by gilesmarshall** [comment]
Ah, I may have confused the issue - $SGAppliance is my own (script-scope) variable that I pass to the cmdlet.  In addtion I've used a different variable for the second example - sorry!

However, the same issue occurs when I pass the FQDN or IP address of the appliance - it prompts for previously supplied data:

`New-SafeguardAsset -Appliance "SG1.LOCAL" -AccessToken $AccessToken -DisplayName "Test6" ....`
`Searching for platforms with 'CentOS Linux Other'`
`Appliance: "SG1.LOCAL"`
`(certificate, local)`
`Provider: local`
`Username: user`
`Invoke-RestMethod : Cannot bind parameter 'Uri'. Cannot convert value "https://"SG1.LOCAL"/RSTS/oauth2/token" to type "System.Uri". Error: "Invalid URI: The hostname could not be parsed."`
`At ...\Modules\safeguard-ps\2.10.399\safeguard-ps.psm1:729 char:28`

BUT...

I've narrowed this down.
1. When I provide the appliance name ***without*** quotes - and then enter the other authentication data as pompted, it successfully creates the asset.
2. If I supply a variable for the appliance name (`$SGappliance` in my examples) I get the error
3. If I **omit** the `-AccessToken $AccessToken` parameter in my original command line, it also successfully creates the asset _without_ any additional prompting for data.

So the problem is that `New-SafeguardAsset` prompts for the appliance info and authentication data when I ***include*** `-AccessToken $AccessToken` parameter.  Which did not occur with 2.9.379.

But I have now learnt that when using `Invoke-SafeguardMethod` I do not need to supply appliance name or access token (unlike when using `Invoke-RestMethod`) so this is a workaround if I edit my scripts accordingly
